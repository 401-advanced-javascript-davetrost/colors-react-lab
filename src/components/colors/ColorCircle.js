import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ColorCircle extends Component {

  static propTypes = {
    backgroundColor: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired
  };
  
  static defaultProps = {
    backgroundColor: 'black',
    backgroundImage: ''
  };

  render() {
    const style = {
      width: '50vw',
      height: '50vh',
      border: 0,
      margin: 'auto',
      borderRadius: '100%',
      backgroundSize: 'contain',
    };

    return (
      <div style={{ ...style, backgroundColor: this.props.backgroundColor, backgroundImage: this.props.backgroundImage }}></div>
    );
  }

}
