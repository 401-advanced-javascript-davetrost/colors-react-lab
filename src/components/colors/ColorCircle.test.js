import React from 'react';
import { shallow } from 'enzyme';
import ColorCircle from './ColorCircle';

describe('Color Circle component', () => {
  it('renders a circle with black background color', () => {
    const wrapper = shallow(      
      <ColorCircle
        backgroundColor='black'
        backgroundImage=''
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders a circle with a background image', () => {
    const wrapper = shallow(      
      <ColorCircle
        backgroundColor='transparent'
        backgroundImage='url(https://i.dailymail.co.uk/i/pix/2016/03/18/15/324D202500000578-3498922-image-a-33_1458315465874.jpg)' 
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
