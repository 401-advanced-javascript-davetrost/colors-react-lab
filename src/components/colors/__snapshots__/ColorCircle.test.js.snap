// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Circle component renders a circle with a background image 1`] = `
<div
  style={
    Object {
      "backgroundColor": "transparent",
      "backgroundImage": "url(https://i.dailymail.co.uk/i/pix/2016/03/18/15/324D202500000578-3498922-image-a-33_1458315465874.jpg)",
      "backgroundSize": "contain",
      "border": 0,
      "borderRadius": "100%",
      "height": "50vh",
      "margin": "auto",
      "width": "50vw",
    }
  }
/>
`;

exports[`Color Circle component renders a circle with black background color 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "backgroundImage": "",
      "backgroundSize": "contain",
      "border": 0,
      "borderRadius": "100%",
      "height": "50vh",
      "margin": "auto",
      "width": "50vw",
    }
  }
/>
`;
